sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -108,23 +109,109 @@ public ServiceBusMessage(BinaryData body) {
      * @param receivedMessage The received message to create new message from.
      *
      * @throws NullPointerException if {@code receivedMessage} is {@code null}.
+     * @throws UnsupportedOperationException if {@link AmqpMessageBodyType} is {@link AmqpMessageBodyType#SEQUENCE} or
+     * {@link AmqpMessageBodyType#VALUE}. See code sample above explaining how to check for {@link AmqpMessageBodyType}
+     * before calling this constructor.
+     * <p><strong>How to check for {@link AmqpMessageBodyType} before calling this constructor</strong></p>
+     * {@codesnippet com.azure.messaging.servicebus.ServiceBusMessage.copyServiceBusMessage}
+     * @throws IllegalStateException for invalid {@link AmqpMessageBodyType}.
+     * @see AmqpAnnotatedMessage
```

Comment:
Why point users to AmqpAnnotatedMessage when we are trying to hide/minimize it's usage?